app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreen.kt

```diff
@@ -241,32 +239,32 @@ fun OtherUserProfileTopBar(
     )
 }
 
-
 @Composable
-fun rememberOtherUserProfileScreenState(
-    snackBarHostState: SnackbarHostState = remember { SnackbarHostState() }
-): OtherUserProfileScreenState {
-    val coroutineScope = rememberCoroutineScope()
-    val clipBoardManager = LocalClipboardManager.current
-
-    return remember {
-        OtherUserProfileScreenState(
-            clipBoardManager = clipBoardManager,
-            snackbarHostState = snackBarHostState,
-            coroutineScope = coroutineScope
-        )
+private fun handleSuccessMessages(
+    snackbarHostState: SnackbarHostState,
+    state: OtherUserProfileState
+) {
+    val requestSentMessage = stringResource(id = R.string.connection_request_sent)
+    LaunchedEffect(state.connectionStatus) {
+        when (val status = state.connectionStatus) {
+            is ConnectionStatus.NotConnected -> if (status.isConnectionRequestPending) snackbarHostState.showSnackbar(requestSentMessage)
+            else -> appLogger.d("Not defined handler for $status")
+        }
     }
 }
 
-class OtherUserProfileScreenState(
-    private val clipBoardManager: ClipboardManager,
-    val snackbarHostState: SnackbarHostState,
-    private val coroutineScope: CoroutineScope
+@Composable
+private fun handleErrorMessages(
+    snackbarHostState: SnackbarHostState,
+    errorState: ErrorState?
 ) {
-
-    fun copy(text: String) {
-        clipBoardManager.setText(AnnotatedString(text))
-        coroutineScope.launch { snackbarHostState.showSnackbar(text) }
+    errorState?.let { errorType ->
+        val message = when (errorType) {
+            is ErrorState.ConnectionRequestError -> stringResource(id = R.string.connection_request_sent_error)
+            else -> stringResource(id = R.string.error_unknown_message)
+        }
+        LaunchedEffect(errorType) {
+            snackbarHostState.showSnackbar(message)
+        }
```

Comment:
since those function have kind of similar behaviour or showing the snackbar , perhaps it makes sense to move the logic to the ViewModel and expose something inside the State object that would have a String resource and UI model would just show the snackbar depending on the string resource ? perhaps then thise piece of code could be moved to ViewModel : 

```
  when (val status = state.connectionStatus) {
            is ConnectionStatus.NotConnected -> if (status.isConnectionRequestPending) 
```

which looks abit like a logic for the ViewModel to me 

let me know what you think :)

Reply:
I agree in the part that maybe we could simplify showing the **error** or **the feature msg** "connection sent" through the snackbar, so I have a middle point on the next PR #579  https://github.com/wireapp/wire-android-reloaded/pull/579/commits/50b23f9d0c854191b2d71bd7b096b02284aa0889 if you can check it out =D 

For this first part, I will just leave it as it is.